### PR TITLE
add support for 'eb --check-conflicts'

### PR DESCRIPTION
**Pull request by boegel**
The code that powers the `check_conflicts` function was taken from the easyconfig test suite, where it has been in use for quite a while already in the tests, and has been successful in detecting conflicts.

When this PR is merged, I'll adjust the easyconfigs test suite to use `check_conflicts` provided by the framework...

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 please review?

**Comment by wpoely86** [comment]
:+1:

**Comment by boegel** [comment]
I've been rethinking this...

The current implementation will *not* report conflicts when running `eb --check-conflicts foss-2016a.eb foss-2015b.eb` because it'll look at each dependency graph in isolation.

I could change the semantics a bit such that the conflict detection is done across dependency graphs for the specified easyconfigs too, which would be more useful I think...

Thoughts?

**Comment by boegel** [comment]
@wpoely86 please rereview?

Now changed to also detect conflicts between listed easyconfig files, e.g.:

```
$ eb --check-conflicts foss-2016a.eb foss-2014b.eb
== temporary log file in case of crash /tmp/eb-MMTI9b/easybuild-nVM0yq.log
Conflict between (dependencies of) easyconfigs: FFTW-3.3.4-gompi-2014b vs FFTW-3.3.4-gompi-2016a
Conflict between (dependencies of) easyconfigs: GCC-4.8.3 vs GCC-4.9.3-2.25
Conflict between (dependencies of) easyconfigs: OpenBLAS-0.2.15-GCC-4.9.3-2.25-LAPACK-3.6.0 vs OpenBLAS-0.2.9-GCC-4.8.3-LAPACK-3.5.0
Conflict between (dependencies of) easyconfigs: OpenMPI-1.10.2-GCC-4.9.3-2.25 vs OpenMPI-1.8.1-GCC-4.8.3
Conflict between (dependencies of) easyconfigs: ScaLAPACK-2.0.2-gompi-2014b-OpenBLAS-0.2.9-LAPACK-3.5.0 vs ScaLAPACK-2.0.2-gompi-2016a-OpenBLAS-0.2.15-LAPACK-3.6.0
Conflict between (dependencies of) easyconfigs: foss-2014b vs foss-2016a
Conflict between (dependencies of) easyconfigs: gompi-2014b vs gompi-2016a
Conflict between (dependencies of) easyconfigs: hwloc-1.11.2-GCC-4.9.3-2.25 vs hwloc-1.9-GCC-4.8.3
Conflict between (dependencies of) easyconfigs: numactl-2.0.11-GCC-4.9.3-2.25 vs numactl-2.0.9-GCC-4.8.3
ERROR: One or more conflicts detected!
```

File: easybuild/tools/robot.py [hunk 1]
```diff
@@ -62,6 +65,75 @@ def det_robot_path(robot_paths_option, tweaked_ecs_path, pr_path, auto_robot=Fal
     return robot_path
 
 
+def check_conflicts(easyconfigs, modtool):
```
Comment (wpoely86) [inline easybuild/tools/robot.py]: doc args

**Comment by boegel** [comment]
Going in, thanks for the review @wpoely86!

**Pull request merged**